### PR TITLE
`DDPNoParametersStrategy`

### PR DESCRIPTION
**Pull request by ordabayevy**
- Remove the default strategy for `OnePassMeanVarStd` and `IncrementalPCA`
- Disable a LightningCLI breaking change introduced in PyTorch Lightning 2.3
- Add `sync_dist=True` flag to train loss log
- Add a new `DDPNoParametersStrategy` #175 
- In `TDigest` use `torchdistributed.gather_object` instead of writing/reading tdigest objects from files

### CODE REVIEW

**Comment by ordabayevy** [comment]
Closed in favor of #206
